eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/NormalQuiescentStateSpec.scala

```diff
@@ -0,0 +1,434 @@
+      alicePeer.fishForMessage(3 seconds) {
+        case Peer.Disconnect(nodeId, _) if nodeId == alice.stateData.asInstanceOf[DATA_NORMAL].commitments.params.remoteParams.nodeId => true
+        case _ => false
+      }
+      disconnect(f)
+    }
+    if (resetConnection || !sendInitialStfu) {
+      // any failure during quiescence will cause alice to disconnect
+      sender.expectMsgType[RES_FAILURE[_, _]]
+      assert(alice.stateData.asInstanceOf[DATA_NORMAL].spliceStatus == SpliceStatus.NoSplice)
+      reconnect(f)
+    }
+
+    // alice sends pending settlement after quiescence ends
+    eventually {
+      c match {
+        case FulfillHtlc => alice2bob.expectMsgType[UpdateFulfillHtlc]
+        case FailHtlc => alice2bob.expectMsgType[UpdateFailHtlc]
+        case FailMalformedHtlc => alice2bob.expectMsgType[UpdateFailMalformedHtlc]
+      }
+    }
+    alice2bob.forward(bob)
+  }
+
+  test("recv stfu when there are pending local changes") { f =>
+    import f._
+    initiateQuiescence(f, sendInitialStfu = false)
+    // we're holding the stfu from alice so that bob can add a pending local change
+    addHtlc(10_000 msat, bob, alice, bob2alice, alice2bob)
+    // bob will not reply to alice's stfu until bob has no pending local commitment changes
+    alice2bob.forward(bob)
+    bob2alice.expectNoMessage(100 millis)
+    crossSign(bob, alice, bob2alice, alice2bob)
+    eventually(assert(bob.stateData.asInstanceOf[ChannelDataWithCommitments].commitments.isIdle))
+    bob2alice.expectMsgType[Stfu]
+    bob2alice.forward(alice)
+    // when both nodes are quiescent, alice will start the splice
+    alice2bob.expectMsgType[SpliceInit]
+    alice2bob.forward(bob)
+    bob2alice.expectMsgType[SpliceAck]
+    bob2alice.forward(alice)
+  }
+
+  test("recv forbidden commands while quiescent") { f =>
+    import f._
+    // alice should reject commands that change the commitment while quiescent
+    val sender1, sender2, sender3 = TestProbe()
+    val cmds = Seq(CMD_ADD_HTLC(sender1.ref, 1_000_000 msat, randomBytes32(), CltvExpiryDelta(144).toCltvExpiry(currentBlockHeight), TestConstants.emptyOnionPacket, None, localOrigin(sender1.ref)),
+      CMD_UPDATE_FEE(FeeratePerKw(100 sat), replyTo_opt = Some(sender2.ref)),
+      CMD_CLOSE(sender3.ref, None, None))
+    initiateQuiescence(f)
+    safeSend(alice, cmds)
+    sender1.expectMsgType[RES_ADD_FAILED[ForbiddenDuringSplice]]
+    sender2.expectMsgType[RES_FAILURE[CMD_UPDATE_FEE, ForbiddenDuringSplice]]
+    sender3.expectMsgType[RES_FAILURE[CMD_CLOSE, ForbiddenDuringSplice]]
+  }
+
+  test("recv fulfill htlc command while initiator awaiting stfu from remote") { f =>
+    receiveSettlementCommand(f, FulfillHtlc, sendInitialStfu = false)
+  }
+
+  test("recv fail htlc command while initiator awaiting stfu from remote") { f =>
+    receiveSettlementCommand(f, FailHtlc, sendInitialStfu = false)
+  }
+
+  test("recv fail malformed htlc command while initiator awaiting stfu from remote") { f =>
```

Comment:
It's honestly overkill to duplicate tests for fail and fail_malformed (both for commands and messages), I think you can remove the malformed ones.

Reply:
Fixed in [1910599](https://github.com/ACINQ/eclair/pull/2680/commits/1910599539cf2852a6b3068b3fbb5d72b49dd020)